python/cudf/cudf/tests/test_udf_masked_ops.py

```diff
@@ -673,3 +672,360 @@ def f(x):
+                "12 ab",
+                "@2a",
+            ]
+        }
+    ],
+)
+def test_string_udf_isalpha(data):
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return st.isalpha()
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "rapids",
+                "123",
+                "",
+                "  ",
+                "12 ab",
+                "@2a",
+            ]
+        }
+    ],
+)
+def test_string_udf_isdigit(data):
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return st.isdigit()
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "rapids",
+                "AI",
+                "",
+                "  ",
+                "12 ab",
+                "@2a",
+                "12.34",
+                "0.123",
+                ".123" ".12abc",
+            ]
+        }
+    ],
+)
+def test_string_udf_isdecimal(data):
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return st.isdecimal()
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "rapids",
+                "AI",
+                "",
+                "rApIdS",
+                "12 ab",
+                "@2a",
+                "12.34",
+                "ABC DEF",
+            ]
+        }
+    ],
+)
+def test_string_udf_isupper(data):
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return st.isupper()
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "rapids",
+                "AI",
+                "",
+                "rApIdS",
+                "12 ab",
+                "@2a",
+                "12.34",
+                "abc def",
+            ]
+        }
+    ],
+)
+def test_string_udf_islower(data):
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return st.islower()
+
+    run_masked_udf_test(func, data, check_dtype=False)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "str_col": [
+                "cudf",
+                "",
+                "  ",
+                "   123   ",
+                "2022 2222",
+                "cuDF",
+            ]
+        }
+    ],
+)
+def test_string_udf_isspace(data):
+    data = cudf.DataFrame(data)
+
+    def func(row):
+        st = row["str_col"]
+        return st.isspace()
+
+    run_masked_udf_test(func, data, check_dtype=False)
```

Comment:
Can we add some explicit tests for lambda functions? Sometimes those have slightly different properties than `def` functions. (Does Numba handle all of this for us, or are we partly responsible for testing this case because of our extensions of Numba support?) For example,

```python
def test_string_udf_isspace_lambda(data):
    data = cudf.DataFrame(data)
    run_masked_udf_test(lambda row: row["str_col"].isspace(), data, check_dtype=False)
```

Reply:
I've not noticed any case where a lambda has any differences from the equivalent `def` function WRT Numba - what properties are you referring to?

Reply:
Lambdas do not have a `__name__` attribute, for example. This is kind of similar to the issue where not all functions have `__code__` or `__defaults__` attributes. https://github.com/rapidsai/cudf/issues/11595

Reply:
Just noting this conversation was unresolved but I think we're deferring to https://github.com/rapidsai/cudf/issues/11595 to handle these kinds of cases. Resolving now.